### PR TITLE
fix: autocomplete suggestion list scroll issue

### LINKED ISSUE #1490: The UI for Mentions is broken when there are many users to mention
https://user-images.githubusercontent.com/3846977/174265552-e047a225-590d-4fa0-ba82-0a1febec06c8.mp4

The relevant component: `package/src/components/AutoCompleteInput/AutoCompleteSuggestionList.tsx`

### PR DESCRIPTION
**Pull request by khushal87**
## 🎯 Goal

Fixes #1490 

## 🎨 UI Changes

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Documentation is updated
- [x] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

File: package/src/components/AutoCompleteInput/AutoCompleteSuggestionList.tsx [hunk 1]
```diff
@@ -31,9 +32,9 @@ export type AutoCompleteSuggestionListPropsWithContext<
 > &
   AutoCompleteSuggestionListComponentProps<StreamChatGenerics>;
 
-const SuggestionsItem: React.FC<TouchableOpacityProps> = (props) => {
-  const { children, ...touchableOpacityProps } = props;
-  return <TouchableOpacity {...touchableOpacityProps}>{children}</TouchableOpacity>;
+const SuggestionsItem: React.FC<PressableProps> = (props) => {
+  const { children, ...pressableProps } = props;
+  return <Pressable {...pressableProps}>{children}</Pressable>;
```
Comment (santhoshvai) [inline package/src/components/AutoCompleteInput/AutoCompleteSuggestionList.tsx:37]: its fine to change to pressable but we remove the opacity change with this 

Look into copying this component https://github.com/mrousavy/react-native-pressable-opacity/blob/master/src/PressableOpacity.tsx

mimics touchable opacity in pressable opacity..

File: package/src/components/AutoCompleteInput/AutoCompleteSuggestionList.tsx [hunk 2]
```diff
@@ -63,62 +66,43 @@ export const AutoCompleteSuggestionListWithContext = <
...
-          );
-        }
-        return null;
+        return (
+          <SuggestionsItem
+            onLayout={(event: LayoutChangeEvent) => {
+              const { height: newHeight } = event.nativeEvent.layout;
+              setItemHeight((prevHeight) => {
+                if (prevHeight !== newHeight) return newHeight;
+                return prevHeight;
```
Comment (santhoshvai) [inline package/src/components/AutoCompleteInput/AutoCompleteSuggestionList.tsx:93]: unnecessary `if (prevHeight !== newHeight) return newHeight;`

primitive values are automatically checked like that by react

**Review by santhoshvai — CHANGES_REQUESTED** [review]
Look into adding opacity anim to pressable or keep touchable opacity pls..

File: package/src/components/AutoCompleteInput/AutoCompleteSuggestionList.tsx (hunk 2 above)
Comment (khushal87) [reply package/src/components/AutoCompleteInput/AutoCompleteSuggestionList.tsx]: Done

File: package/src/components/AutoCompleteInput/AutoCompleteSuggestionList.tsx (hunk 1 above)
Comment (khushal87) [reply package/src/components/AutoCompleteInput/AutoCompleteSuggestionList.tsx]: Done

**Review by santhoshvai — APPROVED** [review]

**Pull request merged**